src/main/java/com/google/sps/servlets/ChatServlet.java

```diff
@@ -0,0 +1,66 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity; 
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableList.Builder;
+import com.google.template.soy.SoyFileSet;
+import com.google.template.soy.tofu.SoyTofu;
+import java.io.File;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/chat")
+public class ChatServlet extends HttpServlet{
+
```

Comment:
Can you pull the hardcoded constants such as "Message" "message-text", etc out into variables?

Reply:
I don't see the constants for this?

Reply:
ahh sorry when I changed the loop for a stream, I deleted the variable. Do you mean pull the message text into a String text = message.getProperty("message-text") like this?

Reply:
The strings themselves should be constants. Like `private static final MESSAGE_TEXT_ATTRIBUTE = "message-text";`